components/hoc/DocumentObjectProvider/DocumentObjectProvider.js

```diff
@@ -0,0 +1,36 @@
+import React, { PureComponent, createContext } from 'react';
+
+export const Context = createContext();
+
+const DocumentObjectProvider = WrappedComponent => {
+  return class extends PureComponent {
+    componentDidMount() {
+      this.forceUpdate(); // force a re-render because we have the document ref now
```

Comment:
I think this is like the first time I've ever seen a legit use for `forceUpdate` 👍